### ISSUE TITLE
Improve Locale.t() to handle context and plurals.

### ISSUE THREAD

**Issue by phaux** [ckeditor/ckeditor5#6158]
## 📝 Provide a description of the improvement

Support more functionality in `Locale.t` function.

* We are lacking support for plurals. The API should be similar to the standard `ngettext`.
* We decided to support passing context as a separate argument (similar to `pgettext`). Right now it looks like passing the context as part of msgid in square brackets is supported.

This will require work in the Locale class and changes in Webpack plugin.

**Comment by phaux** [comment]
## Summary of the new Locale API

Our new API was inspired a lot by Jed and gettext.js which are themselves based on the original GNU gettext approach.

We use the following functions for getting messages:
- `t( 'id', [ ...values ] )` – Nothing changed here. Works the same way as regular `gettext` function but has the placeholder interpolation built-in to not break backwards compatibility. 
- `ct( 'ctx', 'id' )` (stands for "context translation") – Allows specifying the message context. Other common name for this function is `pgettext()`
- `ctq( 'ctx', 'singular id', 'plural id', quantity )` (stands for "context translation quantity") – Used for handling plural forms. Also known as `npgettext()`

The new functions (`ct` and `ctq` don't return strings but rather an object with a few methods (similar to Jed):
- `toString()` – It's there so that the object can be used like it was a regular string.
- `format( ...values )` – Performs the string interpolation. Replaces `%0-9` placeholders with given values.
- `formatToParts( ...values )` – Similar to [`Intl​.List​Format​.formatToParts()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ListFormat/formatToParts). Does the string interpolation, but returns an array of all the literal and dynamic parts of the message.

[Read more about Jed API](https://messageformat.github.io/Jed/) for comparison.

## Changes to translations JSON format

```json
{
  "pl": {
    "PLURAL_FORMS": "nplurals=3; plural=(n == 1) ? 0 : (n < 5) ? 1 : 2",
    "message id": ["localised message", "localised message plural 1", "localised message plural 2"],
    "message context|message id": "localised message"
  }
}
```

Things to note:
- A special key exists for the "plural-forms" header.
- Pipe character separates context and message ID.
- For plural messages, the value is an array of all the plural forms.
- The message IDs and contexts are lowercased.

## Breaking changes 💔

There was functionality for providing the message context in the `t` function as part of the message ID. The syntax was `t( 'message id [context: message context]' )`. 

We decided to remove that functionality because it wasn't used anywhere. A better way of providing more information for the translator are `// translators:` comments. They don't make the message unique though.

**Comment by ma2ciek** [comment]
The issue was closed by ckeditor/ckeditor5-dev#614.
